include/dataformat.hxx

```diff
@@ -72,6 +72,12 @@ class DataFormat {
   virtual bool setGlobalOrigin(int x = 0, int y = 0, int z = 0) = 0; 
   virtual bool setLocalOrigin(int x = 0, int y = 0, int z = 0, int offset_x = 0, int offset_y = 0, int offset_z = 0);
   virtual bool setRecord(int t) = 0; // negative -> latest
+
+  // Add a variable to the file
+  virtual bool addVar_int(const string &name, bool repeat) = 0;
+  virtual bool addVar_BoutReal(const string &name, bool repeat) = 0;
+  virtual bool addVar_Field2D(const string &name, bool repeat) = 0;
+  virtual bool addVar_Field3D(const string &name, bool repeat) = 0;
```

Comment:
Being pedantic, these should not really have underscores. I don't have particularly strong feelings about it though.